app/Services/Model/Imp/SummitService.php

```diff
@@ -4273,4 +4274,28 @@ public function getEventForStreamingInfo(Summit $summit, Member $current_user, i
         return $event;
 
     }
+
+    /**
+     * @param Summit $summit
+     * @param string $badge_qr_code
+     * @return SummitAttendeeBadge
+     * @throws ValidationException
+     */
+    public function validateBadge(Summit $summit, string $badge_qr_code): SummitAttendeeBadge
+    {
+        Log::debug(
+            sprintf("SummitService::validateBadge summit %s badge qr code %s", $summit->getId(), $badge_qr_code));
+
+        $qr_code = SummitAttendeeBadge::decodeQRCodeFor($summit, $badge_qr_code);
+
+        $qr_code_components = explode($summit->getQRRegistryFieldDelimiter(), $qr_code);
```

Comment:
this is not the rite approach please check https://github.com/OpenStackweb/summit-api/blob/3650e40f53f6d2f3c0678460a2d45646df4a6763/app/Services/Model/Imp/SponsorUserInfoGrantService.php#L109
use $fields          = SummitAttendeeBadge::parseQRCode($qr_code); and add valdiations like
https://github.com/OpenStackweb/summit-api/blob/3650e40f53f6d2f3c0678460a2d45646df4a6763/app/Services/Model/Imp/SponsorUserInfoGrantService.php#L121

Reply:
done